credentials/java/com/google/auth/Credentials.java

```diff
@@ -70,6 +72,14 @@ public String getUniverseDomain() throws IOException {
     return GOOGLE_DEFAULT_UNIVERSE;
   }
 
+  public CredentialType getCredentialType() {
+    return this.credentialType;
+  }
+
+  public void setCredentialType(CredentialType credentialType) {
+    this.credentialType = credentialType;
+  }
```

Comment:
Hmm, I wish we could make the setter package-private but it looks like it's in a different module and it needs to be public.

I guess the other option would be to expose this CredentialType stuff only for Oauth2Credentials. That probably leaves a bunch of classes like ApiKeys needing to hold their own implementation.

I think it's probably fine. Maybe we can find another way...